src/simulations/tests/simulator_test.py

```diff
@@ -210,8 +231,42 @@ def test_get_sketch_different_runs_different_random_state(self):
         sketch_estimator_config=sketch_estimator_config)
     df, _ = sim()
     self.assertNotEqual(
-        df.loc[df['run_index'] == 0, 'estimated_cardinality'].values,
-        df.loc[df['run_index'] == 1, 'estimated_cardinality'].values)
+        df.loc[df['run_index'] == 0, 'estimated_cardinality_1'].values,
+        df.loc[df['run_index'] == 1, 'estimated_cardinality_1'].values)
+
+  def test_extend_histogram(self):
+    self.assertEqual(Simulator._extend_histogram(None, [], 1), [0])
+    self.assertEqual(Simulator._extend_histogram(None, [3, 2, 1], 1), [3])
+    self.assertEqual(Simulator._extend_histogram(None, [3, 2, 1], 2), [3, 2])
+    self.assertEqual(Simulator._extend_histogram(None, [3, 2, 1], 3), [3, 2, 1])
+    self.assertEqual(Simulator._extend_histogram(None, [3, 2, 1], 5), [3, 2, 1, 0, 0])
 
+  def test_multiple_frequencies(self):
+    sketch_estimator_config = SketchEstimatorConfig(
+        name='exact-set-multiple-frequencies',
+        sketch_factory=ExactMultiSet,
+        estimator=LosslessEstimator(),
+        max_frequency=3)
+    set_generator_factory = (
+        FakeSetGenerator.get_generator_factory(
+          [[1, 1, 1, 2, 2, 3], [1, 1, 1, 3, 3, 4]]))
+    sim = Simulator(
+        num_runs=1,
+        set_generator_factory=set_generator_factory,
+        sketch_estimator_config=sketch_estimator_config)
+    df, _ = sim()
+    expected_columns = ['num_sets', 'estimated_cardinality_1', 'estimated_cardinality_2',
+                        'estimated_cardinality_3', 'true_cardinality_1',
+                        'true_cardinality_2', 'true_cardinality_3', 'run_index',
+                        'relative_error_1', 'relative_error_2', 'relative_error_3']
+    expected_data = [
+      [1, 3, 2, 1, 3, 2, 1, 0, 0., 0., 0.],
+      [2, 4, 3, 2, 4, 3, 2, 0, 0., 0., 0.]
```

Comment:
Nit. Do we indent by two or four spaces in this context?

Reply:
Done